### PR TITLE
Adding ability to create new pull request.

### PR DESCRIPTION
**Pull request by ps2**
These changes allow the creation of a new pull request using the Octokit package.

I tried following the existing code patterns as best I could. Let me know if you have any feedback. Thanks!

### CODE REVIEW

File: OctoKit/PullRequest.swift [hunk 1]
```diff
@@ -323,6 +366,37 @@ enum PullRequestRouter: JSONPostRouter {
                 parameters["maintainer_can_modify"] = (mantainerCanModify ? "true" : "false")
             }
             return parameters
+
+//            func pullRequest(_ session: RequestKitURLSession = URLSession.shared,
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:370]: This seems obsolete?

File: OctoKit/PullRequest.swift [hunk 2]
```diff
@@ -323,6 +366,37 @@ enum PullRequestRouter: JSONPostRouter {
...
+                "base": base
+            ]
+            if let headRepo {
+                parameters["head_repo"] = headRepo
+            }
+            if let body {
+                parameters["body"] = body
+            }
+            if let mantainerCanModify {
+                parameters["maintainer_can_modify"] = (mantainerCanModify ? "true" : "false")
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:394]: Did you try to just put the swift bool here? Codable should be able to figure it out…

File: OctoKit/PullRequest.swift [hunk 3]
```diff
@@ -323,6 +366,37 @@ enum PullRequestRouter: JSONPostRouter {
...
+            }
+            if let draft {
+                parameters["draft"] = (draft ? "true" : "false")
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:397]: Same here. Just writing it like this should be sufficient.
```suggestion
                parameters["draft"] = draft
```

File: OctoKit/PullRequest.swift [hunk 4]
```diff
@@ -83,6 +83,45 @@ open class PullRequest: Codable {
...
+     - parameter draft: Indicates whether the pull request is a draft.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    @discardableResult
+    func pullRequest(_ session: RequestKitURLSession = URLSession.shared,
+                     owner: String,
+                     repository: String,
+                     title: String,
+                     head: String,
+                     head_repo: String? = nil,
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:106]: Please stick to camel case

File: OctoKit/PullRequest.swift [hunk 5]
```diff
@@ -83,6 +83,45 @@ open class PullRequest: Codable {
...
 
 public extension Octokit {
+    /**
+     Create a pull request
+     - parameter session: RequestKitURLSession, defaults to URLSession.shared
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repository: The name of the repository.
+     - parameter title: The title of the new pull request.
+     - parameter head: The name of the branch where your changes are implemented.
+     - parameter head_repo: The name of the repository where the changes in the pull request were made.
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:93]: Please stick to camel case

**Review by pietbrauer — CHANGES_REQUESTED** [review]
Thanks a lot! Please also add the async/await compatible version.

**Review by pietbrauer — CHANGES_REQUESTED** [review]
Thanks for the changes! Something I totally forgot yesterday was that some tests would also be really nice.

**Comment by ps2** [comment]
Added tests, thanks!

**Review by pietbrauer — APPROVED** [review]

File: OctoKit/PullRequest.swift [hunk 6]
```diff
@@ -320,7 +397,27 @@ enum PullRequestRouter: JSONPostRouter {
...
+            }
+            return parameters
+
+        case let .createPullRequest(_, _, _, title, head, headRepo, base, body, mantainerCanModify, draft):
+            var parameters: [String: Any] = [
+                "title": title,
+                "head": head,
+                "base": base
+            ]
+            if let headRepo {
```
Comment (pietbrauer) [inline OctoKit/PullRequest.swift:410]: Tests are failing because in the swift version we support this shorthand is not available.

**Review by pietbrauer — DISMISSED** [review]

**Comment by pietbrauer** [comment]
@ps2 Linting is still failing sadly otherwise it's good to go.

**Review by pietbrauer — APPROVED** [review]

**Pull request merged**